File: LEETCODE/EASY/DAY 3/TWO SUM-III/TWO SUM III.py
class TwoSum:
    def __init__(self):
        self.table = {}
    
    def add(self, number: int) -> None:
        self.table[number] = self.table.get(number, 0) + 1
    
    def find(self, value: int) -> bool:
        for num in self.table:
            y = value - num
            if y == num:
                if self.table[num] >= 2:
                    return True
            elif y in self.table:
                return True
        return False

# Example Usage
ts = TwoSum()
ts.add(1)
ts.add(3)
ts.add(5)
print(ts.find(4))  # True
print(ts.find(7))  # False

File: LEETCODE/EASY/DAY 03/TWO SUM III.java
class TwoSum {
    private Map<Integer, Integer> table = new HashMap<>();
    
    // Add number to the data structure
    public void add(int number) {
        int count = table.getOrDefault(number, 0);
        table.put(number, count + 1);
    }
    
    // Find if there exists a pair with given sum
    public boolean find(int value) {
        for (Map.Entry<Integer, Integer> entry : table.entrySet()) {
            int num = entry.getKey();
            int y = value - num;
            if (y == num) {
                if (entry.getValue() >= 2) return true;
            } else if (table.containsKey(y)) {
                return true;
            }
        }
        return false;
    }
}
